src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageSolutionDetailControlModel.cs

```diff
@@ -127,30 +143,35 @@ private async Task UpdateInstalledVersionsAsync(CancellationToken cancellationTo
             {
                 try
                 {
-                    IPackageReferenceContextInfo installedVersion = await GetInstalledPackageAsync(project.NuGetProject, Id, cancellationToken);
-                    if (installedVersion != null)
+                    (IPackageReferenceContextInfo TopLevelPackage, IPackageReferenceContextInfo TransitivePackage) installedPackage = await GetInstalledPackageAsync(project.NuGetProject, Id, cancellationToken);
+                    var topLevelPackageVersion = installedPackage.TopLevelPackage;
+                    var transitivePackageVersion = installedPackage.TransitivePackage;
+
+                    project.RequestedVersion = null;
+                    project.InstalledVersion = null;
+                    project.PackageLevel = null;
+                    project.AutoReferenced = false;
+                    project.InstalledVersionMaxVulnerability = -1;
+
+                    if (transitivePackageVersion != null)
                     {
-                        project.InstalledVersion = installedVersion.Identity.Version;
-                        installedVersionsSet.Add(installedVersion.Identity.Version);
-                        project.AutoReferenced = installedVersion.IsAutoReferenced;
-
-                        if (project.NuGetProject.ProjectStyle.Equals(ProjectStyle.PackageReference))
-                        {
-                            project.RequestedVersion = installedVersion?.AllowedVersions?.OriginalString;
-                        }
-
-                        if (_searchResultPackage.VulnerableVersions.TryGetValue(installedVersion.Identity.Version, out int vulnerable))
-                        {
-                            project.InstalledVersionMaxVulnerability = vulnerable;
-                            vulnerabilitiesSet.Add(installedVersion.Identity.Version);
-                        }
+                        project.PackageLevel = "Transitive";
```

Comment:
We should use the PackageLevel enum from NuGet.VisualStudio.Common.

Reply:
One of the UX board comments was that it should be the same text as the packages list headers, and unfortunately it is "Top-level" instead of "Top-Level" and if I use the enum is going to be with capital L.

Do you want me to change the header to have capital L? Which one do you think looks better?

Reply:
It's the View's responsibility to decide what string should be displayed based on the PackageLevel type so we should use the enum in the ViewModel (which this class acts as, despite its name), and convert to a string resource in the View.

Reply:
This string also needs to be localized.